src/libraries/System.Text.Json/src/Resources/Strings.resx

```diff
@@ -810,4 +809,22 @@
   <data name="Argument_InvalidOffLen" xml:space="preserve">
     <value>Offset and length were out of bounds for the array or count is greater than the number of elements from index to the end of the source collection.</value>
   </data>
-</root>
+  <data name="CannotWriteWithinString" xml:space="preserve">
+    <value>Cannot write the requested JSON property or value until the final string value segment has been written.</value>
+  </data>
+  <data name="Arg_MustBeRune" xml:space="preserve">
+    <value>Object must be of type Rune.</value>
+  </data>
+  <data name="ArgumentOutOfRange_IndexMustBeLess" xml:space="preserve">
+    <value>Index was out of range. Must be non-negative and less than the size of the collection.</value>
+  </data>
+  <data name="Argument_DestinationTooShort" xml:space="preserve">
+    <value>Destination is too short.</value>
+  </data>
+  <data name="Argument_CannotExtractScalar" xml:space="preserve">
+    <value>Cannot extract a Unicode scalar value from the specified index in the input.</value>
+  </data>
+  <data name="CannotMixEncodings" xml:space="preserve">
+    <value>Cannot mix encodings between string value segments. The previous segment's encoding was '{0}' and the current segment's encoding is '{1}'.</value>
```

Comment:
Consider updating the wording here (and in the previous messages) to use "not supported" instead of "cannot":
```suggestion
    <value>Mixing UTF encodings in a single multi-segment JSON string is not supported. The previous segment's encoding was '{0}' and the current segment's encoding is '{1}'.</value>
```